### PR TITLE
[TextFields] Updates to SimpleTextFIeld and InputChipView prototype

### PR DESCRIPTION
**Pull request by andrewoverton**
This PR contains a lot of updates to SimpleTextField. Some of these updates are for little things I noticed I got wrong after looking at the guidelines a bit more. The more substantial changes introduced here, however, are related to feedback given in the design doc overview meeting in January. As a reminder, the two takeaways from that meeting were:

1. SimpleTextField shouldn't depend on MDCContainerScheme
2. The styles should not be represented as enum cases. Instead, they should be objects.

These changes address those pieces of feedback. While I still need to get around to updating the design doc to reflect the changes, I don't see the harm in merging the updates into their home in a MaterialComponentsExamples experimental directory.

This PR also contains changes that are not directly related to SimpleTextField!

Late last year, when we had the week of free time to do whatever we wanted, I began thinking about incorporating chips into SimpleTextFields. Not far into it I decided that any view implementing input chip functionality should be its own view, i.e. it shouldn't be stuffed in a UITextField, which carries inheritance-related burden. I started work on such a view in January, and briefly used it during the SimpleTextField design doc overview meeting to demonstrate how SimpleTextField styling could be generalized across views.

Following that meeting, and following the refactoring work to address feedback on SimpleTextFields, I updated the InputChipView to see how extensible my "container style as object" approach was. I think the work here, while definitely not a finished product, demonstrates that it's fairly usable. MDCSimpleTextField (I added an MDC) and InputChipView (no MDC) both conform to a protocol called MDCContainedInputView. That protocol declares a property for objects conforming to MDCContainerStyling. There are a few other classes and protocols in there, but I will write or talk more about those in a different setting. The key here is that I think it all might work, based off these changes. There are still some loose ends, but they can be taken care of. They _will_ be taken care of if the team decides to take this direction. They maybe won't if we decide to continue investing in MDCTextFields for all of our user input needs. If we do end up deciding to bring all this into the codebase I want the next addition to this family of contained input views to be a subclass of UITextView, for multiline text input.

This gif shows the updated MDCSimpleTextField styled as an "outlined" text field:
stf_outlined_2
This gif shows the updated MDCSimpleTextField styled as a "filled" text field:
stf_filled_2
This gif shows the updated InputChipView styled as an "outlined" input chip view whose chips do not wrap:
outlined_non_wrapping_2

This gif shows the updated InputChipView styled as an "outlined" input chip view whose chips wrap:
outlined_wrapping_2
This gif shows the updated InputChipView styled as a "filled" input chip view whose chips do not wrap:
filled_non_wrapping_2
This gif shows the updated InputChipView styled as a "filled" input chip view whose chips wrap:
filled_wrapping_2

### CODE REVIEW

**Comment by codeman7** [comment]
GIFs look great!

File: components/TextFields/examples/experimental/InputChipViewExampleViewController.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
```
Comment (jverkoey) [inline components/TextFields/examples/experimental/InputChipViewExampleViewController.h]: Please update copyright dates here and throughout.

**Review by jverkoey — APPROVED** [review]
LGTM for experimental. Keep in mind that in order for this code to land in `src/` it will need to be reviewed as a sequence of smaller PRs.

**Comment by jverkoey** [comment]
Please address the build failures prior to checking this in.

**Pull request merged**